src/isomorphic/classic/types/ReactPropTypes.js

```diff
@@ -116,10 +116,13 @@ function createChainableTypeChecker(validate) {
     if (props[propName] == null) {
       var locationName = ReactPropTypeLocationNames[location];
       if (isRequired) {
-        return new Error(
-          `Required ${locationName} \`${propFullName}\` was not specified in ` +
-          `\`${componentName}\`.`
-        );
+        var message = `Required ${locationName} \`${propFullName}\` was not specified in ` +
+          `\`${componentName}\`.`;
+        var lowerPropName = propName.toLowerCase();
+        if (props[lowerPropName] != null) {
```

Comment:
This typecheck seems wrong to me.  To detect improper capitalization, I feel like you need to normalize both sides (eg. send both of them toLowerCase).  I think this code as-is will only catch the case where the user happens to use all lowercase.